src/eduid_idp/mischttp.py

```diff
@@ -317,8 +319,10 @@ def set_cookie(name, path, logger, config, value=''):
     :type value: string
     :rtype: bool
     """
+    if six.PY3 and type(value) == bytes:
+        value = value.decode('utf-8')
     cookie = cherrypy.response.cookie
-    cookie[name] = base64.b64encode(str(value))
+    cookie[name] = b64encode(value)
```

Comment:
The `value` arg to b64encode, in python3, needs to be a byte string, but 2 lines above (if PY3), you are decoding it to a unicode object (str) in case it is a byte string.